pkgs/development/compilers/openjdk/22.nix

```diff
@@ -254,6 +255,14 @@ stdenv.mkDerivation (finalAttrs: {
   pos = __curPos;
   meta = import ./meta.nix lib featureVersion;
 
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/java";
+  versionCheckProgramArg = "${./JavaGetVersionTest.java}";
```

Comment:
Why isn't `java --version` not good?

Reply:
right, `java --version` is good enough, I'm removing the `JavaGetVersionTest.java` file.